core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -726,6 +726,31 @@ public SqlNode toSql(@Nullable RexProgram program, RexNode rex) {
           }
         }
         elseNode = caseNodeList.get(caseNodeList.size() - 1);
+
+        if (caseCall.getType().getSqlTypeName() == SqlTypeName.BOOLEAN
```

Comment:
For other databases, this is a special implementation. Can we rewrite CASE WHEN in OracleSqlDialect? Maybe `OracleSqlDialect.unparseCall` is a good place.

Reply:
I wanted to design it this way at first, but I couldn't get the return type of SqlCase in unparseCall, and this conversion has to ensure that the return type is a boolean.

Reply:
+1